lib/improver/cli/nbhood_iterate_with_mask.py

```diff
@@ -132,25 +128,32 @@ def main(argv=None):
     args = parser.parse_args(args=argv)
 
     # Load Cubes
-    cube = load_cube(args.input_filepath)
-    mask_cube = load_cube(args.input_mask_filepath)
-    weights = load_cube(args.weights_for_collapsing_dim) if \
+    cube = cli.inputcube(args.input_filepath)
+    mask_cube = cli.inputcube(args.input_mask_filepath)
+    weights = cli.inputcube(args.weights_for_collapsing_dim) if \
         args.collapse_dimension else None
 
     # Process Cube
-    result, intermediate_cube = process(
-        cube, mask_cube, weights, args.coord_for_masking, args.radius,
-        args.radii_by_lead_time, args.sum_or_fraction, args.re_mask,
-        args.collapse_dimension)
-
-    # Save Cube
-    save_netcdf(result, args.output_filepath)
-    if args.intermediate_filepath is not None:
-        save_netcdf(intermediate_cube, args.intermediate_filepath)
-
-
-def process(cube, mask_cube, weights, coord_for_masking, radius=None,
-            radii_by_lead_time=None, sum_or_fraction="fraction", re_mask=False,
+    # pylint: disable=E1123
+    process(cube, mask_cube, weights, coord_for_masking=args.coord_for_masking,
+            radius=args.radius, radii_by_lead_time=args.radii_by_lead_time,
+            sum_or_fraction=args.sum_or_fraction, re_mask=args.re_mask,
+            collapse_dimension=args.collapse_dimension,
+            output=args.output_filepath,
+            intermediate_output=args.intermediate_filepath)
+
+
+@cli.clizefy
+@cli.with_output
+@cli.with_intermediate_output
+def process(cube: cli.inputcube,
+            mask: cli.inputcube,
+            weights: cli.inputcube = None,
+            *,
+            coord_for_masking, radius: float = None,
+            radii_by_lead_time=None,
+            sum_or_fraction="fraction",
+            re_mask=False,
```

Comment:
I think  `--re-mask` looks confusing as an option, we could change that name to single word `remask`.